cmd/device-worker/main.go

```diff
@@ -106,12 +107,19 @@ func main() {
 	wl.RegisterObserver(dataMonitor)
 	configManager.RegisterObserver(dataMonitor)
 	dataMonitor.Start()
+
+	metricsStore, err := metrics.NewTSDB(dataDir)
+	configManager.RegisterObserver(metricsStore)
+
+	if err != nil {
```

Comment:
Kind of - if the device doesn't start before it's registered, it won't be seen as a CR, if it doesn't start after it has been registered and rebooted the last seen time in the device status wouldn't be updated.

Reply:
does the error here cause worker to fail? it seems we just log. I thought ondra added a way to report errors back to operator.

Reply:
Yes - the trick is that `log.Fatal` logs _and_ terminates the application.
The events are sent as part of the heartbeats and if the worker is not initialized, they wouldn't be sent (and event if, that would cause the last time seen be updated on the edge device CR - which is not true, because the worker is crashing).

Reply:
wow. I didn't know that. thanks.